javascript/ql/lib/semmle/javascript/frameworks/SQL.qll

```diff
@@ -540,3 +598,145 @@ private module SpannerCsv {
     }
   }
 }
+
+/**
+ * Provides classes modeling the Google Cloud Spanner library.
+ */
+private module Spanner {
```

Comment:
This entire module is mostly copy-paste from the model that was "deprecated" by models-as-data.  

You could try to use models-as-data, by extending the rows in `SpannerSinks` with new rows, where you invent some new sink-type (like `database-access`).  
In `AdditionalSources.qll` you can then access the new sinks like: `ModelOutput::getASinkNode("database-access").getARhs()`.   

Models-as-data has an `Awaited` path token that you can use. 

The rest looks fine. 

Another comment: I usually don't squash commits after un-drafting a PR. Having multiple commits makes it easier to review.

Reply:
> You could try to use models-as-data, by extending the rows in SpannerSinks with new rows, where you invent some new sink-type (like database-access).

You mean create a new `SpannerSources` module, right? I've been looking at the MaD work (the details of which I didn't yet know all that well), and it looks promising. I'll start working on that now. Thank you.

> Another comment: I usually don't squash commits after un-drafting a PR. Having multiple commits makes it easier to review.

Thanks for pointing that out, I'll happily match what others are doing :) Is it customary to do a final squash before merging?

Reply:
> You mean create a new `SpannerSources` module, right? I've been looking at the MaD work (the details of which I didn't yet know all that well), and it looks promising. I'll start working on that now. Thank you.

Yes, right. 
Probably just a new class `class SpannerSources extends ModelInput::SourceModelCsv`. 
And then use it like: `ModelOutput::getASourceNode("database-access").getAnImmediateUse()`. 

> Thanks for pointing that out, I'll happily match what others are doing :) Is it customary to do a final squash before merging? 

No it's not. 
The GitHub web UI shows a link from a commit to the PR that included the PR.  
That feature makes squash commits somewhat redundant.

Reply:
I just pushed commit 09a28c428c25714b1b10cd4312c5e989152833e6. Looks good to you?

> No it's not.
The GitHub web UI shows a link from a commit to the PR that included the PR.
That feature makes squash commits somewhat redundant.

Alright, will do :) Thanks for pointing that out.